src/secret_sharing/malicious_replicated.rs

```diff
@@ -140,38 +146,50 @@ impl<F: Field> Mul<F> for MaliciousReplicated<F> {
     }
 }
 
+#[async_trait]
 impl<F: Field> Downgrade for MaliciousReplicated<F> {
     type Target = Replicated<F>;
-    fn downgrade(self) -> UnauthorizedDowngradeWrapper<Self::Target> {
+    async fn downgrade(self) -> UnauthorizedDowngradeWrapper<Self::Target>
+    where
+        Self::Target: Send,
+    {
         UnauthorizedDowngradeWrapper(self.x)
     }
 }
 
+#[async_trait]
 impl<T, U> Downgrade for (T, U)
 where
-    T: Downgrade,
-    U: Downgrade,
+    T: Downgrade + Send,
+    U: Downgrade + Send,
 {
-    type Target = (<T as Downgrade>::Target, <U as Downgrade>::Target);
-    fn downgrade(self) -> UnauthorizedDowngradeWrapper<Self::Target> {
-        UnauthorizedDowngradeWrapper((
-            self.0.downgrade().access_without_downgrade(),
-            self.1.downgrade().access_without_downgrade(),
-        ))
+    type Target = (<T>::Target, <U>::Target);
+    async fn downgrade(self) -> UnauthorizedDowngradeWrapper<Self::Target>
+    where
+        Self::Target: Send,
+    {
+        let one = self.0.downgrade().await.access_without_downgrade();
```

Comment:
self.0 is giving error : `<T as Downgrade>::Target` cannot be sent between threads safely
the trait `std::marker::Send` is not implemented for `<T as Downgrade>::Target`rustc[Click for full compiler diagnostic](rust-analyzer-diagnostics-view:/diagnostic%20message?3#file%3A%2F%2F%2FUsers%2Frichaj%2FProposals%2FIPA%2Fraw-ipa%2Fsrc%2Fsecret_sharing%2Fmalicious_replicated.rs)
malicious_replicated.rs(171, 26): required by a bound introduced by this call
malicious_replicated.rs(27, 23): required by a bound in `Downgrade::downgrade`
malicious_replicated.rs(167, 23): consider further restricting the associated type: `, <T as Downgrade>::Target: std::marker::Send`

Reply:
join()?

Reply:
This is still not happy still downgrade is async and access_without_downgrade is where we join
`<U as Downgrade>::Target` is not a future
within `futures::future::Join<<T as Downgrade>::Target, <U as Downgrade>::Target>`, the trait `futures::Future` is not implemented for `<U as Downgrade>::Target`
<U as Downgrade>::Target must be a future or must implement `IntoFuture` to be awaited
the return type of a function must have a statically known size